Second Part Of Diploma

kosalexkos opened the pull request:
First commit  of the second part of diploma project

kosalexkos committed:
Second Part Initial Commit

First commit  of the second part of diploma project

kosalexkos committed:
Update EventServiceImpl.java

comparison issues

kosalexkos committed:
Update docker-compose.yml

docker-compose issues

kosalexkos committed:
docker issues

tried to fix issues with docker stat container

kosalexkos committed:
docker issues

kosalexkos committed:
Fixing Stat Issues

Fixed the response issues in stat server

felixoff commented on ewm-service/src/main/java/ru/practicum/category/controller/CategoryAdminController.java:
```diff
@@ -0,0 +1,36 @@
+        return categoryService.patch(catId, categoryDto);
+    }
+
+    @DeleteMapping("/{catId}")
```
Одинаковые части путей можно вынести в отдельную переменную для избежания дубляжей

felixoff commented on ewm-service/src/main/java/ru/practicum/event/controller/EventPrivateController.java:
```diff
@@ -0,0 +1,65 @@
+        return eventService.patchByInitiator(userId, eventId, updateEventUserRequest);
+    }
+
+    @PatchMapping("/{eventId}/requests")
```
То же замечание про одинаковые части путей

felixoff commented on ewm-service/src/main/java/ru/practicum/exception/ApiError.java:
```diff
@@ -0,0 +1,20 @@
+    private HttpStatus status;
+    private String reason;
+    private String message;
+    private StackTraceElement[] errors;
```
Вместо множественных private лучше использовать @FieldDefaults

kosalexkos committed:
After review

Changed path variables in the controllers to optimize the usage of memory and deleted the @Autowired annotations as they are redundant
